### PR TITLE
Weighted average optimized

### PR DESCRIPTION
**Pull request by sepandhaghighi**
#### Reference Issues/PRs
#274 
#### What does this implement/fix? Explain your changes.
- `weighted_average` method optimized
- `pycmAverageError` class added
- Document modified

### CODE REVIEW

**Comment by sepandhaghighi** [comment]
@sadrasabouri @alirezazolanvari

Hi folks ;-)

This PR is for #274 , please check carefully and give me some feedback ;-)

Notice 1 : Do not hurry, we have enough time ;-)
Notice 2 : This PR has 2 reviewer, so please don't merge this PR to `dev` branch before finishing the discussion

Best Regards
Sepand Haghighi

File: pycm/pycm_obj.py [hunk 1]
```diff
@@ -656,24 +656,40 @@ def relabel(self, mapping):
...
 
         :param item: a class item for which weighted average has to be calculated
-        :type item1:str
-        :param weights: Explicitly passes weights or Support or P
-        :type item2:dict
-    
+        :type item:str
+        :param weight: Explicitly passes weights or Support or P
+        :type weight:dict
+        :param none_omit: none items omit flag
```
Comment (sadrasabouri) [inline pycm/pycm_obj.py:668]: `none items omitting flag` isn't better?

File: pycm/pycm_param.py [hunk 2]
```diff
@@ -39,6 +39,10 @@
 
 COMPARE_RESULT_WARNING = "Confusion matrices are too close and the best one can not be recognized."
 
+AVERAGE_WEIGHT_ERROR = "The weight type must be dictionary and also must be set for all classes."
```
Comment (alirezazolanvari) [inline pycm/pycm_param.py:42]: I think it's better to be written as follows:

"The weight type must be dictionary and also must be specified for all of the classes."

File: pycm/pycm_obj.py [hunk 3]
```diff
@@ -656,24 +656,40 @@ def relabel(self, mapping):
         self.FP = self.class_stat["FP"]
         self.FN = self.class_stat["FN"]
         __class_stat_init__(self)
-    def weighted_average(self, item, weights=None):
+
+    def weighted_average(self, param, weight=None, none_omit=False):
         """
-        Calculate Weighted average of input parameter.
+        Calculate weighted average of input parameter.
```
Comment (alirezazolanvari) [inline pycm/pycm_obj.py:662]: It should be written as follows:

"Calculate the weighted average of the input parameter"

**Pull request merged**